### DISCUSSION TITLE
v3.2.x

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#4129] [Releases]
The release manager for `v3.2.x` is @lbdreyer

The release feature branch is [v3.2.x](https://github.com/SciTools/iris/tree/v3.2.x).

| Minor Release | Status | Milestone | Project | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.2.0rc0` | Completed | [v3.2.0](https://github.com/SciTools/iris/milestone/44) | [Iris v3.2.0](https://github.com/orgs/SciTools/projects/2/views/4) | 29/01/2022 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.2.0rc0/) | [v3.2.0rc0](https://github.com/SciTools/iris/releases/tag/v3.2.0rc0) | [Available](https://anaconda.org/conda-forge/iris?version=3.2.0rc0&channel=rc_iris) | - |
| `v3.2.0` | Completed | [v3.2.0](https://github.com/SciTools/iris/milestone/44) | [Iris v3.2.0](https://github.com/orgs/SciTools/projects/2/views/4) | 15/02/2022 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.2.0.post0/whatsnew/latest.html) | [v3.2.0](https://github.com/SciTools/iris/releases/tag/v3.2.0.post0) | [Available](https://anaconda.org/conda-forge/iris?version=3.2.0.post0&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.2.0.post0/) |

| Patch Release | Status | Milestone | Project | Date | RTD | Release | conda-forage | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.2.1` | Completed | - | - | 11/03/2022 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.2.1/) | [v3.2.1](https://github.com/SciTools/iris/releases/tag/v3.2.1) | [Available](https://anaconda.org/conda-forge/iris?version=3.2.1&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.2.1/) |

**Comment by jamesp** [comment]
Ahead of the game! Nice.  Anyone fancy taking on release manager for this release @SciTools/iris-devs ?

**Reply by bjlittle** [reply]
Couldn't agree more! :tada: 

This is a really great opportunity to shape and improve our release process and documentation.

Tempted but unclear about what's involved? Then checkout our [Release](https://scitools-iris.readthedocs.io/en/latest/developers_guide/release.html) documentation.

Still not convinced? Well, we can always organise an experienced release buddy to help you seal the deal and get the release across the line!

**Reply by bjlittle** [reply]
@lbdreyer Fancy taking on `iris` 3.2.x as the release manager? :smile:

**Reply by lbdreyer** [reply]
Sure, I'd be up for that!

**Reply by wjbenfold** [reply]
Can I shadow you releasing 3.2 @lbdreyer?

**Reply by lbdreyer** [reply]
> Can I shadow you releasing 3.2 @lbdreyer?

Sure!

**Comment by lbdreyer** [comment]
Below is a plan for the next few weeks:

### How we will work together:

- Work will be managed using a Github project board (link tbc - I will convert the current board to a new Projects beta board, for now use [this board](https://github.com/SciTools/iris/projects/32))
- We will not use the milestone
- We will put issues on the project board and link the associated PR to the issue. If there is no issue, put the PR on the board
- "Stop starting/Start Stopping" i.e. see if there is something you can help to finish before starting something new
- Identify a "buddy" to help you (e.g. to review) when you start working on something
- Add yourself and your buddy as the "assignee" on the issue/PR
- Use this github discussion thread to discuss work on Iris 3.2 (e.g. if you are picking something up and looking for a buddy)


### General plan of what to do:
- Prioritise the "MVP" work. This includes everything above the MVP divider. Note this work may increase once the MVP for supporting unstructured meshes has been agreed
- Then the next priority will be finishing off what is already in progress
- After that we will address what's above the "good to have" divider. Note I have roughly tried to prioritise this, but you may disagree with the priority if you have a better understanding of the context. If so you're welcome to move it higher up in priority
- We will discuss bigger items (e.g. #3413) once the MVP (specifically the unstructured meshes work) has been firmed up. After which we will have a better understanding of whether they will fit in.

**Reply by trexfeathers** [reply]
>- We will put issues on the project board and link the associated PR to the issue. If there is no issue, put the PR on the board

Several of us had a look today and as far as we can tell this is not yet a feature of the Kanban Board in the New Projects Beta.

**Reply by lbdreyer** [reply]
It's very odd!  [This update from last month](https://github.blog/changelog/2021-12-16-the-new-github-issues-12-16-update/) suggests that linked PRs should be displayed but I don't see them. Perhaps we should raise this as feedback to Github.
For now, the best I can suggest is to either look at the "Linked Pull Requests" column in the ["Priorities" view](https://github.com/orgs/SciTools/projects/2/views/1) or to go to the issue itself

**Comment by lbdreyer** [comment]
### Cutting the v3.2.x branch

My expectation is that we will create the v3.2.x branch just before the we cut the release candidate. This follows the process we have used for previous releases. So for now all PRs should target the master branch. 

A consequence of this is that anything that gets merged to the master branch in the next 3 weeks will be included in v3.2, so please keep this in mind before merging; we want to avoid merging something which is only partly complete and which won't be finished before 3.2 release (we have been stung by this in the past!)

**Comment by lbdreyer** [comment]
### v3.2.0 Final Release
The plan is to cut the v3.2.0 release tomorrow.
I have not seen any particular feedback of issues users have encountered with Iris 3.2.0rc0 so I believe we are good to go.
If there is anything last minute (and small!) you would like to get in, please do it today.

**Reply by wjbenfold** [reply]
https://github.com/SciTools/iris/issues/4582 has been raised, though I'm not sure how much of the issue lies in Iris and how much in Cartopy

**Reply by wjbenfold** [reply]
It's no worse in 3.2.0rc0 than in previous releases, but not fully fixed

**Reply by lbdreyer** [reply]
Thanks for raising @wjbenfold! Considering it's a bit close to the deadline, let's instead try to get that in for a bug release (3.2.1)
